edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/AuthChainHelpers.cs

```diff
@@ -27,6 +27,18 @@ public static Option<string> GetAuthTarget(Option<string> authChain)
             return authChainIds.FirstOption(id => true);
         }
 
+        public static Option<string> GetAuthParent(Option<string> authChain)
+        {
+            return authChain.Match(
+                chain =>
+                {
+                    string[] authChainIds = GetAuthChainIds(chain);
+                    // The auth target is second element after the target
+                    return authChainIds.Skip(1).FirstOption(id => true);
```

Comment:
Check if chain has >0 elements?

Reply:
What is parent used for?

Reply:
it's not needed to check because skip will return empty collection and then parent is None.
it's used for making the onbehalfof call  to refresh the identity